datastore/modelassertmanager.go

```diff
@@ -130,6 +293,43 @@ func (db *DB) UpdateModelAssert(m ModelAssertion) error {
 	return nil
 }
 
+// UpsertSignedModelAssert creates or updates signed model assertion
+func (db *DB) UpsertSignedModelAssert(modelID int, assertion asserts.Assertion) error {
```

Comment:
we really need to make sure that we aren't storing an assertion with a lower revision than the current one, this is a central invariant of assertion behavior

Reply:
Currently, we just override the entry in the database without checking the revision. In the UI user can increase or decrease the revision without any checks. 
One possibility to fix this is to remove the revision field from the UI and increment it in the backend before storing the assertion in the database.

Reply:
yes, I think not letting the user set a revision seems correct.  It doesn't make sense from an assertion pov to create a different version (different values) at an older revision point